sympy/integrals/heurisch.py

```diff
@@ -732,7 +732,7 @@ def find_non_syms(expr):
             raise PolynomialError
         solution = solve_lin_sys(numer.coeffs(), coeff_ring, _raw=False)
 
-        if solution is None:
+        if solution is None or any(s.has_free(*V) for s in solution.values()):
```

Comment:
Maybe the `solve_lin_sys` function should be changed to handle this case differently. This is the only place in all of SymPy that actually uses the function.

Reply:
> This is the only place in all of SymPy that actually uses the function.

I don't quite understand why; the `solve_lin_sys` function is quite commonly used because it is the internal implementation of `sympy.linsolve`. And this is not a problem with `solve_lin_sys`. The solution it gives is correct, but it is not applicable in this case.